### PR TITLE
mailio: add version 0.24.1

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **mailio/0.24.1**

#### Motivation
There are several improvements since 0.23.0.

#### Details
https://github.com/karastojko/mailio/compare/0.23.0...0.24.1

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/mailio/all/conanfile.py [hunk 1]
```diff
@@ -27,61 +27,38 @@ class MailioConan(ConanFile):
...
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
-            raise ConanInvalidConfiguration(
-                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
-            )
+        check_min_cppstd(self, 17)
 
     def build_requirements(self):
         self.tool_requires("cmake/[>=3.16.3 <4]")
+        if Version(self.version) >= "0.24.0" and \
```
Comment (jcar87) [inline recipes/mailio/all/conanfile.py:47]: I believe this needs to be in a `validate()` or `validate_build()` method?

Comment (jcar87) [reply recipes/mailio/all/conanfile.py:47]: However, if the error was this:

```
/Users/jenkins/workspace/cci_prod_PR-25955/conan-home/p/b/maili77426e52e76ce/b/src/src/percent.cpp:37:25: error: implicit instantiation of undefined template 'std::vector<std::basic_string<char>>'
vector<string> percent::encode(const string& txt, const string& charset) const
                        ^
/Applications/conan/xcode/13.0/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/iosfwd:216:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector;
                           ^
/Users/jenkins/workspace/cci_prod_PR-25955/conan-home/p/b/maili77426e52e76ce/b/src/src/percent.cpp:39:20: error: implicit instantiation of undefined template 'std::vector<std::basic_string<char>>'
    vector<string> enc_text;
                   ^
/Applications/conan/xcode/13.0/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/iosfwd:216:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector;
                           ^
```

there's a chance the fix is just including `<vector>` in the right place

Comment (karastojko) [reply recipes/mailio/all/conanfile.py]: The [version 0.24.1](https://github.com/karastojko/mailio/releases/tag/0.24.1) with the proposed fix ix ready.

Comment (toge) [reply recipes/mailio/all/conanfile.py]: @karastojko 
Thanks a lot!

**Review by AbrilRBS — APPROVED** [review]
Thanks!

**Review by ErniGH — APPROVED** [review]
LGTM

**Pull request merged**